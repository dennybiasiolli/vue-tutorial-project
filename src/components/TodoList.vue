<script setup>
import TodoItem from './TodoItem.vue'

const props = defineProps({
  title: {
    type: String,
    required: true
  },
  items: {
    type: Array,
    required: true,
  }
})
</script>

<template>
  <h4>{{ props.title }}</h4>
  <ul>
    <li
      v-for="(item, index) in props.items"
      :key="item.id"
      :class="{ 'item-completed': item.completed }"
    >
      {{ index + 1 }}.
      <TodoItem
        :text="item.text"
        :completed="item.completed"
      />
    </li>
  </ul>
</template>
