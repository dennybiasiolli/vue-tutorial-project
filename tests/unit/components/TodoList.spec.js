import { shallowMount } from '@vue/test-utils'
import { expect, it } from 'vitest'
import TodoList from '@/components/TodoList.vue'

it('should render as expected', () => {
  const wrapper = shallowMount(TodoList, {
    props: {
      title: 'my to do list',
      items: [
        { id: 123, text: 'my todo 1', completed: false },
        { id: 456, text: 'my todo 2', completed: false },
        { id: 789, text: 'my todo 3', completed: true },
      ],
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
})
