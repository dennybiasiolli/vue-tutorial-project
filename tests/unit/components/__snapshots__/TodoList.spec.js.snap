// Vitest Snapshot v1

exports[`should render as expected 1`] = `
"<h4>my to do list</h4>
<ul>
  <li class=\\"\\">1. <todo-item-stub text=\\"my todo 1\\" completed=\\"false\\"></todo-item-stub>
  </li>
  <li class=\\"\\">2. <todo-item-stub text=\\"my todo 2\\" completed=\\"false\\"></todo-item-stub>
  </li>
  <li class=\\"item-completed\\">3. <todo-item-stub text=\\"my todo 3\\" completed=\\"true\\"></todo-item-stub>
  </li>
</ul>"
`;
